### ISSUE TITLE
error: target not found: - flags add empty strings to update command

### ISSUE THREAD

**Issue by jakebailey** [rmarquis/pacaur#319]
This may be related to #318.

When running `pacaur -Syu` with other options (for example, `--devel --noedit --needed`), those flags seem to get added to the `pacman -Syu` call as empty strings.

For example (bash -x dump, up until AUR stuff which is fine):
```
~ ❯❯❯ bash -x pacaur -Syu --devel --noedit --needed
+ version=4.2.21
+ LC_COLLATE=C
+ source /etc/makepkg.conf
++ DLAGENTS=('ftp::/usr/bin/curl -fC - --ftp-pasv --retry 3 --retry-delay 3 -o %o %u' 'http::/usr/bin/curl -fLC - --retry 3 --retry-delay 3 -o %o %u' 'https::/usr/bin/curl -fLC - --retry 3 --retry-delay 3 -o %o %u' 'rsync::/usr/bin/rsync --no-motd -z %u %o' 'scp::/usr/bin/scp -C %u %o')
++ VCSCLIENTS=('bzr::bzr' 'git::git' 'hg::mercurial' 'svn::subversion')
++ CARCH=x86_64
++ CHOST=x86_64-unknown-linux-gnu
++ CPPFLAGS=-D_FORTIFY_SOURCE=2
++ CFLAGS='-march=core-avx2 -O2 -pipe'
++ CXXFLAGS='-march=core-avx2 -O2 -pipe'
++ LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro
++ MAKEFLAGS=-j8
++ DEBUG_CFLAGS='-g -fvar-tracking-assignments'
++ DEBUG_CXXFLAGS='-g -fvar-tracking-assignments'
++ BUILDENV=(!distcc color !ccache check !sign)
++ OPTIONS=(strip docs !libtool !staticlibs emptydirs zipman purge !upx !debug)
++ INTEGRITY_CHECK=(md5)
++ STRIP_BINARIES=--strip-all
++ STRIP_SHARED=--strip-unneeded
++ STRIP_STATIC=--strip-debug
++ MAN_DIRS=({usr{,/local}{,/share},opt/*}/{man,info})
++ DOC_DIRS=(usr/{,local/}{,share/}{doc,gtk-doc} opt/*/{doc,gtk-doc})
++ PURGE_TARGETS=(usr/{,share}/info/dir .packlist *.pod)
++ COMPRESSGZ=(gzip -c -f -n)
++ COMPRESSBZ2=(bzip2 -c -f)
++ COMPRESSXZ=(xz -T0 -c -z -)
++ COMPRESSLRZ=(lrzip -q)
++ COMPRESSLZO=(lzop -q)
++ COMPRESSZ=(compress -c -f)
++ PKGEXT=.pkg.tar.xz
++ SRCEXT=.src.tar.gz
+ [[ -r /home/jake/.makepkg.conf ]]
+ tmpdir=/tmp/pacaurtmp-jake
+ builddir=/tmp/pacaurtmp-jake
+ editor=vim
+ editpkgbuild=true
+ editinstall=true
+ fallback=true
+ clean=true
+ cleandevel=true
+ sortbyvotes=true
+ sudoloop=false
+ pacmanbin=pacman
+ [[ -n '' ]]
+ configdir=/etc/xdg/pacaur
+ userconfigdir=/home/jake/.config/pacaur
+ source /etc/xdg/pacaur/config
++ builddir=/home/jake/pacaurtmp
++ editor=vim
++ cleandevel=false
+ [[ -r /home/jake/.config/pacaur/config ]]
+ [[ -d /tmp/pacaurtmp-jake ]]
+ [[ ! -w /tmp/pacaurtmp-jake ]]
+ [[ ! -d /tmp/pacaurtmp-jake ]]
+ [[ ! -d /home/jake/pacaurtmp ]]
+ trap ctrlc INT
+ args=("$@")
+ for i in '"${args[@]}"'
+ [[ -Syu =~ ^-[a-zA-Z0-9] ]]
+ opts+=("$i")
+ for i in '"${args[@]}"'
+ [[ --devel =~ ^-[a-zA-Z0-9] ]]
+ for i in '"${args[@]}"'
+ [[ --noedit =~ ^-[a-zA-Z0-9] ]]
+ for i in '"${args[@]}"'
+ [[ --needed =~ ^-[a-zA-Z0-9] ]]
+ count=0
+ [[ -n -Syu ]]
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ pacS=1
+ operation=sync
+ coweropts+=("-f")
+ [[ -Syu =~ w ]]
+ installpkg=true
+ [[ -Syu =~ g ]]
+ [[ -Syu =~ l ]]
+ [[ -Syu =~ p ]]
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ [[ -n 1 ]]
+ operation=sync
+ refresh=true
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n 1 ]]
+ continue
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ [[ -n '' ]]
+ [[ -n 1 ]]
+ operation=sync
+ upgrade=true
+ coweropts+=("-f")
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n 1 ]]
+ continue
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ case "$OPTARG" in
+ devel=true
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ case "$OPTARG" in
+ noedit=true
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ case "$OPTARG" in
+ needed=true
+ pacopts+=("--needed")
+ makeopts+=("--needed")
+ getopts sidmykufecqrahvVDQRSTUbglnoptw-: OPT
+ [[ -z '' ]]
+ break
++ grep '^Color' /etc/pacman.conf
+ [[ -n Color ]]
+ [[ '' != \n\e\v\e\r ]]
+ [[ '' = \a\l\w\a\y\s ]]
+ coweropts+=("--color=auto")
+ reset='\e[0m'
+ colorR='\e[1;31m'
+ colorG='\e[1;32m'
+ colorY='\e[1;33m'
+ colorB='\e[1;34m'
+ colorM='\e[1;35m'
+ colorW='\e[1;39m'
+ pacmanarg=("${pacmanarg[@]/--/}")
+ pacmanarg=("${pacmanarg[@]/-r/}")
+ pacmanarg=("${pacmanarg[@]/-a/}")
+ [[ sync = sync ]]
+ [[ ! -n '' ]]
+ [[ ! -n '' ]]
+ [[ ! -n '' ]]
+ [[ 1000 -eq 0 ]]
+ [[ -n 1 ]]
+ pacmanarg=("${pacmanarg[@]/-e/}")
+ [[ -n 1 ]]
+ [[ -n '' ]]
+ [[ -n 1 ]]
+ [[ -n '' ]]
+ [[ ! -n vim ]]
+ [[ ! -w /home/jake/pacaurtmp ]]
+ [[ -z '' ]]
+ [[ sync = download ]]
+ [[ sync = sync ]]
+ [[ ! -n true ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ case $operation in
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n true ]]
+ [[ ! -n '' ]]
+ ClassifyPkgs
+ [[ true = true ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ ! -n '' ]]
+ [[ ! -n '' ]]
+ [[ -n '' ]]
+ repopkgs=($(LC_COLLATE=C comm -23 <(tr ' ' '\n' <<< ${noaurpkgs[@]} | LC_COLLATE=C sort -u) <(tr ' ' '\n' <<< ${aurpkgs[@]} | LC_COLLATE=C sort -u)))
++ LC_COLLATE=C
++ comm -23 /dev/fd/63 /dev/fd/62
+++ tr ' ' '\n'
+++ LC_COLLATE=C
+++ sort -u
+++ tr ' ' '\n'
+++ LC_COLLATE=C
+++ sort -u
+ unset pkgs
+ [[ ! -n '' ]]
+ sudo pacman -S -y -u '' '' '' --needed
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
error: target not found: 
error: target not found: 
error: target not found: 
+ [[ ! -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ ! -n '' ]]
+ [[ true = true ]]
+ Core
+ getignoredpkgs
+ ignoredpkgs+=($(grep '^IgnorePkg' '/etc/pacman.conf' | cut -d '=' -f 2- | tr -d "'\""))
++ grep '^IgnorePkg' /etc/pacman.conf
++ cut -d = -f 2-
++ tr -d ''\''"'
+ [[ -e /home/jake/.config/cower/config ]]
+ ignoredpkgs=("${ignoredpkgs[@]//,/ }")
+ [[ -n true ]]
+ UpgradeAur
+ note i '\e[1;39mStarting AUR upgrade...\e[0m'
+ case "$1" in
+ echo -e '\e[1;34m::\e[0m \e[1;39mStarting AUR upgrade...\e[0m'
```

If I change the flags to have fewer (like just `--devel`), the number of empty strings added is smaller. No extra flags works (of course).

**Comment by rmarquis** [comment]
Thanks, I can reproduce. Likely related to the cleanup commit. Will have a look asap.

**Comment by rmarquis** [comment]
I reverted the faulty commit in 0700e285eef17c88963249f2c07c66c30ca9e5de and readded the important fixes in 17ce8c8d5dac36f8778442089858e158c061f956.

Should be fixed pacaur-git. Will make a new release shortly.
